cdap-ui/app/cdap/components/AbstractWidget/SqlSelectorWidget/index.tsx

```diff
@@ -0,0 +1,332 @@
+/*
+ * Copyright © 2019 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the 'License'); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an 'AS IS' BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import isEqual from 'lodash/isEqual';
+import classnames from 'classnames';
+
+import withStyles, { WithStyles } from '@material-ui/core/styles/withStyles';
+import Button from '@material-ui/core/Button';
+
+import ThemeWrapper from 'components/ThemeWrapper';
+import SchemaContainer from 'components/AbstractWidget/SqlSelectorWidget/SchemaContainer';
+import If from 'components/If';
+
+export const styles = () => {
+  return {
+    root: {
+      padding: '4px',
+      display: 'flex',
+      'flex-direction': 'column',
+    },
+    buttonContainer: {
+      display: 'flex',
+      justifyContent: 'flex-end',
+      '> button': {
+        marginLeft: '5px',
+      },
+    },
+    errorMessage: {
+      margin: '5px',
+    },
+    emptyMessage: {
+      margin: '0px',
+    },
+  };
+};
+
+interface ISqlSelectorWidgetProps extends WithStyles<typeof styles> {
+  value: string;
+  inputSchema: any;
+  onChange: (arg0: any) => void;
+}
+
+export interface IFieldSchema {
+  selected: boolean;
+  name: string;
+  alias: string;
+}
+export interface IParsedInputSchema {
+  key: string;
+  name: string;
+  expanded: boolean;
+  schema: IFieldSchema[];
+}
+interface ISqlSelectorWidgetState {
+  expandAll: boolean;
+  errors: { stageCount: object; message: string; exist: boolean };
+  parsedInputSchemas: IParsedInputSchema[];
+  aliases: object;
+  modelCopy: any;
+}
+
+class SqlSelectorWidgetView extends React.PureComponent<
+  ISqlSelectorWidgetProps,
+  ISqlSelectorWidgetState
+> {
+  public state = {
+    expandAll: false,
+    errors: {
+      stageCount: {},
+      message: 'Please create one or more aliases for duplicate field names.',
+      exist: false,
+    },
+    parsedInputSchemas: null,
```

Comment:
I am not sure if this is valid. I think this will make the type inference of `parsedInputSchemas` in the class as `any` right?